core/gather/driver/storage.js

```diff
@@ -36,27 +36,30 @@ const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
 /**
  * @param {LH.Gatherer.ProtocolSession} session
  * @param {string} url
+ * @param {Array<string>} [resetOriginStorage]
  * @return {Promise<LH.IcuMessage[]>}
  */
-async function clearDataForOrigin(session, url) {
+async function clearDataForOrigin(session, url, resetOriginStorage) {
   const status = {msg: 'Cleaning origin data', id: 'lh:storage:clearDataForOrigin'};
   log.time(status);
 
   const warnings = [];
 
   const origin = new URL(url).origin;
 
-  // Clear some types of storage.
-  // Cookies are not cleared, so the user isn't logged out.
-  // indexeddb, websql, and localstorage are not cleared to prevent loss of potentially important data.
-  //   https://chromedevtools.github.io/debugger-protocol-viewer/tot/Storage/#type-StorageType
-  const typesToClear = [
-    // 'cookies',
+  /** Clear some types of storage.
+   Cookies are not cleared, so the user isn't logged out.
+  indexeddb, websql, and localstorage are not cleared to prevent loss of potentially important data.
+  https://chromedevtools.github.io/debugger-protocol-viewer/tot/Storage/#type-StorageType
+  */
+  const defaultTypesToClear = [
     'file_systems',
     'shader_cache',
     'service_workers',
     'cache_storage',
-  ].join(',');
+  ]
+
+  const typesToClear = (resetOriginStorage ?? defaultTypesToClear).join(',')
```

Comment:
We should always clear `defaultTypesToClear` so we should concat these two lists. That way if `resetOriginStorage` is set to `[]` we will still clear the default set of storage locations.

Reply:
That makes sense, unless we want to give users the ability to not clear certain types of cache but allow others.

For example, maybe they want to clear the HTTP cache but want to keep service workers.  idk 🤷‍♀️, for my usecase I'll either use `["all"]` or use the default.

Reply:
What do you think?

Reply:
> For example, maybe they want to clear the HTTP cache but want to keep service workers. idk 🤷‍♀️, for my usecase I'll either use ["all"] or use the default.

We discussed this and you make a valid point. We should have a new config option like `clearStorageTypes` that takes an array of strings like you suggest. The default value of this array will be the current set of types that we clear.

For reference, default settings are defined here: https://github.com/GoogleChrome/lighthouse/blob/main/core/config/constants.js#L92